network/ocsp.go

```diff
@@ -124,6 +125,47 @@ const (
 	OcspHTTPClientDefaultTimeout = time.Second * time.Duration(15)
 )
 
+// NewOCSPConfigByName return initialized OCSPConfig config using flags registered with RegisterCertVerifierArgsForService
+func NewOCSPConfigByName(flags *flag.FlagSet, name string, namerFunc NamerFunc) (*OCSPConfig, error) {
+	var url, required, fromCert string
+	var checkOnlyLeafCert bool
+	if f := flags.Lookup(namerFunc(name, "url", "ocsp")); f != nil {
+		url = f.Value.String()
+		if url == "" {
+			url = tlsOcspURL
+		}
+	}
+	if f := flags.Lookup(namerFunc(name, "required", "ocsp")); f != nil {
+		required = f.Value.String()
+		if required == "" {
+			required = tlsOcspRequired
+		}
+	}
+	if f := flags.Lookup(namerFunc(name, "from_cert", "ocsp")); f != nil {
+		fromCert = f.Value.String()
+		if fromCert == "" {
+			fromCert = tlsOcspFromCert
+		}
+	}
+	if f := flags.Lookup(namerFunc(name, "check_only_leaf_certificate", "ocsp")); f != nil {
+		getter, ok := f.Value.(flag.Getter)
+		if !ok {
+			log.Fatal("Can't cast flag's Value to Getter")
+		}
+		val, ok := getter.Get().(*bool)
+		if val == nil {
+			checkOnlyLeafCert = false
+		} else {
+			if !ok {
```

Comment:
Hmm, is it even possible when the getter returns `not-nil` **value**  but the `ok` will be false?

Reply:
hm. you are right. actually is not. but if we will leave:
```
val, _ := getter.Get().(*bool)
		if val == nil {
			checkOnlyLeafCert = false
		} else {
			checkOnlyLeafCert = *val
		}
```
as for me, it will not be so obvious after several month why it did so. And I will need spend a time to understand that it is legal)